lib/gateway/Shard.js

```diff
@@ -1258,6 +1260,64 @@ class Shard extends EventEmitter {
                 this.emit("messageReactionRemoveEmoji", message, packet.d.emoji);
                 break;
             }
+            case "MESSAGE_POLL_VOTE_ADD": {
+                const user = this.client.users.get(packet.d.user_id);
+                const channel = this.client.getChannel(packet.d.channel_id);
+                let message;
+                if(channel) {
+                    message = channel.messages.get(packet.d.message_id);
+                }
+                if(!message) {
+                    message = {
+                        id: packet.d.message_id,
+                        channel: channel || {id: packet.d.channel_id}
+                    };
+                    if(packet.d.guild_id) {
+                        message.guildID = packet.d.guild_id;
+                        if(!message.channel.guild) {
+                            message.channel.guild = {id: packet.d.guild_id};
+                        }
+                    }
+                }
+                /**
+                 * Fired when a user votes on a poll. If the poll allows multiple selection, one event will be sent per answer
+                 * @event Client#messagePollVoteAdd
+                 * @prop {Message | Object} message The message object. If the message is not cached, this will be an object with `id` and `channel` keys. If the channel is not cached, channel key will be an object with an id and a `guild` key with only an id. No other property is guaranteed
+                 * @prop {User | Object} user The user who voted on the poll. If the user is not cached, this will be an object with only an `id` key
+                 * @prop {Number} answerID The ID of the answer
+                 */
+                this.emit("messagePollVoteAdd", message, user, packet.d.answer_id);
```

Comment:
Can we add/remove answer counts if the message is cached please

Reply:
Yeah nw. That should be okay 👍🏻